File: C++/C++Basic/06.Conditional-Statements-Advanced-Exercise/08.onTimeForTheExam.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

int main()
{
    int hourOfExam, minutesOfExam, hourOfArrival, minutesOfArrival;
    cin >> hourOfExam >> minutesOfExam >> hourOfArrival >> minutesOfArrival;

    int totalMinutesExam = hourOfExam * 60 + minutesOfExam;
    int totalMinutesStudent = hourOfArrival * 60 + minutesOfArrival;

    int difference = totalMinutesExam - totalMinutesStudent;
    bool isLate = false;

    if (difference < 0)
    {
        isLate = true;
        cout << "Late" << endl;
    }
    else if (difference >= 0 && difference <= 30)
    {
        cout << "On time" << endl;
    }
    else if (difference > 30)
    {
        cout << "Early" << endl;
    }

    int absoluteDifference = abs(difference);
    int hours = absoluteDifference / 60;
    int minutes = absoluteDifference % 60;

    if (isLate)
    {
        if (hours >= 1)
        {
            if (minutes < 10)
            {
                cout << hours << ":0" << minutes << " hours after the start" << endl;
            }
            else
            {
                cout << hours << ":" << minutes << " hours after the start" << endl;
            }
        }
        else
        {
            cout << minutes << " minutes after the start" << endl;
        }
    }
    else
    {
        if (absoluteDifference > 0)
        {
            if (hours > 0)
            {
                if (minutes < 10)
                {
                    cout << hours << ":0" << minutes << " hours before the start" << endl;
                }
                else
                {
                   cout << hours << ":" << minutes << " hours before the start" << endl;
                }
            }
            else
            {
               cout << minutes << " minutes before the start" << endl;
            }
        }
    }


    return 0;
}

File: Python/Basic/06.Conditional-Statements-Advanced-Exercise/08.on_time_for_the_exam.py
exam_hour = int(input())
exam_min = int(input())
hour_of_arrival = int(input())
min_of_arrival = int(input())

total_minutes_exam = exam_hour * 60 + exam_min
total_minutes_student = hour_of_arrival * 60 + min_of_arrival

min_diff = abs(total_minutes_exam - total_minutes_student)

hour = 0
minute = 0

if total_minutes_student > total_minutes_exam:
    print('Late')

    if min_diff < 60:
        print(f"{min_diff} minutes after the start")
    else:
        hour = min_diff // 60
        minute = min_diff % 60

        if minute < 10:
            print(f"{hour}:0{minute} hours after the start")
        else:
            print(f"{hour}:{minute} hours after the start")

elif total_minutes_student < total_minutes_exam - 30:
    print("Early")

    if min_diff < 60:
        print(f"{min_diff} minutes before the start")
    else:
        hour = min_diff // 60
        minute = min_diff % 60

        if minute < 10:
            print(f"{hour}:0{minute} hours before the start")
        else:
            print(f"{hour}:{minute} hours before the start")

else:
    print("On time")
    print(f"{min_diff} minutes before the start")